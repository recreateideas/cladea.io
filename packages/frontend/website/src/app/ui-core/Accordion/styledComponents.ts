import styled, { css } from 'styled-components';
import {
    Accordion as MUIAccordion,
    AccordionDetails as MUIAccordionDetails,
    AccordionSummary as MUIAccordionSummary,
} from '@mui/material';

export const Container = styled.div``;
export const Accordion = styled(MUIAccordion)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            background-color: ${palette.neutrals.secondaryBg};
            box-shadow: none;
            background-image: none;
        `;
    }}
`;
export const AccordionDetails = styled(MUIAccordionDetails)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            padding: 0px ${padding.small} ${padding.small} ${padding.small};
        `;
    }}
`;
export const AccordionSummary = styled(MUIAccordionSummary)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    sizes,
                    layout: { padding },
                },
            },
        } = props;
        return css`
            &.Mui-expanded {
                min-height: ${sizes.accordion.medium};
            }
            > div {
                margin: ${padding.small} 0 !important;
            }
            padding: 0px ${padding.small};
        `;
    }}
`;
export const Title = styled.div``;
