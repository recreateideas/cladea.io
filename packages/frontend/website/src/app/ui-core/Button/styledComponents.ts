import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
