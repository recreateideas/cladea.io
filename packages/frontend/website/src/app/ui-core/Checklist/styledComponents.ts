import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
