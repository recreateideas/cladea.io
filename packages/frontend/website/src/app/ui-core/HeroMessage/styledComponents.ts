import styled, { css } from 'styled-components';

export const Container = styled.div<{ type: 'error' | 'warn' | 'success' }>`
    ${(props) => {
        const {
            theme: {
                dsl: { hexToRgba, styles, layout, palette },
            },
            type,
        } = props;
        const colorMap: {
            [type: string]: {
                color: string;
                backgroundColor: string;
            };
        } = {
            error: {
                color: palette.fonts.error,
                backgroundColor: hexToRgba(palette.secondary.orange[500], 10),
            },
            success: {
                color: palette.fonts.success,
                backgroundColor: hexToRgba(palette.secondary.green[500], 10),
            },
            warn: {
                color: palette.fonts.warn,
                backgroundColor: hexToRgba(palette.secondary.yellow[500], 10),
            },
        };
        return css`
            ${colorMap[type]}
            white-space: pre;
            padding: ${layout.padding.small};
            border-radius: ${styles.borderRadius};
            overflow-x: auto;
            .section {
                display: flex;
                .icon {
                    margin-right: ${layout.margin.small};
                }
            }
        `;
    }}
`;
