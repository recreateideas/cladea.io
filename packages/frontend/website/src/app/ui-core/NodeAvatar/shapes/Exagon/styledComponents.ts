import styled, { css } from 'styled-components';

interface IExagon {
    inner?: boolean;
    backgroundColor: string;
    width: number;
    height: number;
    border: number;
}

const exagon = ({ inner, width, height, border, backgroundColor }: IExagon) => {
    return css`
        ${inner &&
        css`
            left: 1px;
            z-index: 1;
        `}
        position: relative;
        height: ${height}px;
        width: ${width}px;
        background-color: ${backgroundColor};
        margin: ${height / 2}px 0;
        &:before,
        &:after {
            content: '';
            position: absolute;
            width: 0;
            border-left: ${width / 2}px solid transparent;
            border-right: ${width / 2}px solid transparent;
        }
        &:before {
            bottom: 100%;
            border-bottom: ${border}px solid ${backgroundColor};
        }
        &:after {
            top: 100%;
            width: 0;
            border-top: ${border}px solid ${backgroundColor};
        }
    `;
};

export const Container = styled.div<{ size?: 'small' | 'medium'; inner?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    palette,
                    // layout,
                },
            },
            inner,
            size = 'medium',
        } = props;
        const sizes = {
            small: 62,
            medium: 102,
        };
        const length = sizes[size];
        const whRatio = 1.732;
        const width = inner ? length - 2 : length;
        const height = inner ? width / whRatio + 1 : width / whRatio;
        const border = inner ? height / 2 - 1 : height / 2;
        const backgroundColor = inner ? palette.neutrals.secondaryBg : palette.neutrals.borders;
        return css`
            ${exagon({ inner, width, height, border, backgroundColor })}
            .content {
                position: relative;
                top: -1px;
                .icon {
                    display: flex;
                    justify-content: center;
                }
                .header {
                    flex-direction: column;
                    text-align: center;
                }
                .react-flow__handle-bottom {
                    top: ${height +
                    border -
                    2 +
                    1}px; // 1px because content is pushed up 1px in line 69
                    z-index: 2;
                }
            }
        `;
    }}
`;
