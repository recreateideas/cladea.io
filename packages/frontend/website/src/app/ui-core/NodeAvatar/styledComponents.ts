import styled, { css, ThemedStyledProps } from 'styled-components';

interface IContainer {
    size?: 'small' | 'medium';
    isActive?: boolean;
    customCss?: ThemedStyledProps<any, any>;
}

export const DefaultContainer = styled.div<IContainer>`
    box-sizing: border-box;
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            isActive,
            size = 'small',
            customCss,
        } = props;
        const sizeMap: {
            [key: string]: {
                height: number | string;
                width: number | string;
            };
        } = {
            small: {
                width: '100%',
                height: 32,
            },
            medium: {
                width: 256,
                height: 56,
            },
        };
        return css`
            ${isActive &&
            css`
                background-color: ${palette.neutrals.tertiaryBg} !important;
            `}
            background-color: ${palette.neutrals.secondaryBg};
            border: solid 1px ${palette.neutrals.borders};
            border-radius: ${styles.borderRadius};
            ${sizeMap[size]}
            ${customCss}
        `;
    }}
`;

export const Header = styled.div`
    display: flex;
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            .left {
                display: flex;
                &.as-column {
                    flex-direction: column;
                }
            }
            padding: ${padding.xsmall};
            display: flex;
            justify-content: space-between;
        `;
    }}
`;

export const HeaderText = styled.div<{ size?: 'small' | 'medium' }>`
    display: flex;
    flex-direction: column;
    ${(props) => {
        const {
            theme: {
                dsl: {
                    typography: { typefaces },
                    layout: { padding },
                },
            },
            size = 'small',
        } = props;
        const sizes = {
            small: 'caption',
            medium: 'p',
        };
        const nameSize = sizes[size];
        return css`
            padding-left: ${padding.xsmall};
            .name {
                ${typefaces[nameSize]({ color: 'primary' })}
            }
            .type {
                ${typefaces.caption()}
            }
        `;
    }}
`;

export const Body = styled.div``;
