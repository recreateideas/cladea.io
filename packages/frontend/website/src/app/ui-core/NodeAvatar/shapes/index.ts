import { Exagon } from './Exagon';

export const shapes = {
    exagon: Exagon,
};
