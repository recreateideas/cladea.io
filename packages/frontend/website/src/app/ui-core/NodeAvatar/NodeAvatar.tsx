import { useState, ReactElement } from 'react';
import { SVGIcon } from '../../ui-core';
import { DefaultContainer, Header, HeaderText, Body } from './styledComponents';
import { shapes } from './shapes';
import { CoreNodeProps } from '@types';

export const NodeAvatar = (props: CoreNodeProps): ReactElement => {
    const {
        size = 'small',
        customCss,
        label,
        type,
        icon,
        className,
        shape,
        actions,
        children,
    } = props;
    const [isActive, setIsActive] = useState<boolean>(false);
    const onMouseEnter = () => {
        setIsActive(true);
    };
    const onMouseLeave = () => {
        setIsActive(false);
    };
    const Container = shape ? shapes[shape] : DefaultContainer;
    const flexDirection = shape === 'exagon' ? 'as-column' : 'as-row';
    return (
        <Container
            className={className}
            {...{ onMouseEnter, onMouseLeave, customCss, isActive, size }}
        >
            <div className="content">
                <Header className="header">
                    <div className={`left ${flexDirection}`}>
                        <div className="icon">
                            <SVGIcon {...{ name: icon, preserveColor: true }} />
                        </div>
                        <HeaderText className="header-text" {...{ size }}>
                            <div className="name">{label}</div>
                            {type && <div className="type">{type}</div>}
                        </HeaderText>
                    </div>
                    {actions && <div className="node-actions">{actions}</div>}
                </Header>
                <Body>{children}</Body>
            </div>
        </Container>
    );
};

NodeAvatar.displayName = 'NodeAvatar';

NodeAvatar.defaultProps = {};
