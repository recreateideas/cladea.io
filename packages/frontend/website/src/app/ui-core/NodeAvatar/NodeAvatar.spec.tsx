import { NodeAvatar } from './NodeAvatar';
import { TestProvider } from 'src/tests';
import { DefaultContainer } from './styledComponents';

const { mountWithProviderAndTheme } = TestProvider();
describe('<NodeAvatar/>', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<NodeAvatar name="some-name" />, state);
        expect(root).toMatchSnapshot();
    });
    it('should render the header fo rexagon correctly', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <NodeAvatar shape="exagon" name="some-name" />,
            state
        );
        expect(root).toMatchSnapshot();
    });
    it('should render the type if it exists', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <NodeAvatar {...{ name: 'some-name', type: 'some-type' }} />,
            state
        );
        const typeDiv = root.find('.type');
        expect(typeDiv.length).toBe(1);
    });
    it('should NOT render the type if DOES NOT exists', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <NodeAvatar {...{ name: 'some-name', type: undefined }} />,
            state
        );
        const typeDiv = root.find('.type');
        expect(typeDiv.length).toBe(0);
    });
    it('should render the active state', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(
            <NodeAvatar
                className="node-avatar-test-container"
                {...{ name: 'some-name', type: undefined }}
            />,
            state
        );
        const container = root.find(DefaultContainer);
        container.simulate('mouseenter');
        root.update();
    });
});
