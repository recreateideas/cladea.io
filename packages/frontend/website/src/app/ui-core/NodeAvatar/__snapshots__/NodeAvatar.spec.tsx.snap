// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NodeAvatar/> should render 1`] = `
Array [
  .c2 {
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
  position: relative;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  box-sizing: border-box;
  background-color: #f9fafb;
  border: solid 1px #dedeeb;
  border-radius: 4px;
  width: 100%;
  height: 32px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 .left {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .left.as-column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 4px;
}

.c3 .name {
  font-family: 'Roboto',sans-serif;
  font-size: 11px;
  line-height: 16px;
  color: #031A51;
  text-transform: initial;
  font-weight: 400;
}

.c3 .type {
  font-family: 'Roboto',sans-serif;
  font-size: 11px;
  line-height: 16px;
  color: #8697BF;
  text-transform: initial;
  font-weight: 400;
}

<div
    className="c0"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="small"
  >
    <div
      className="content"
    >
      <div
        className="c1 header"
      >
        <div
          className="left as-row"
        >
          <div
            className="icon"
          >
            <div
              className="c2 svg-icon alert  "
            >
              <svg
                height="20px"
                width="20px"
              >
                ico-alert.svg
              </svg>
            </div>
          </div>
          <div
            className="c3 header-text"
            size="small"
          >
            <div
              className="name"
            />
          </div>
        </div>
      </div>
      <div
        className=""
      />
    </div>
  </div>,
  "",
]
`;

exports[`<NodeAvatar/> should render the header fo rexagon correctly 1`] = `
Array [
  .c3 {
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
  position: relative;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 .left {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 .left.as-column {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 4px;
}

.c4 .name {
  font-family: 'Roboto',sans-serif;
  font-size: 11px;
  line-height: 16px;
  color: #031A51;
  text-transform: initial;
  font-weight: 400;
}

.c4 .type {
  font-family: 'Roboto',sans-serif;
  font-size: 11px;
  line-height: 16px;
  color: #8697BF;
  text-transform: initial;
  font-weight: 400;
}

.c0 {
  position: relative;
  height: 35.79676674364896px;
  width: 62px;
  background-color: #dedeeb;
  margin: 17.89838337182448px 0;
}

.c0:before,
.c0:after {
  content: '';
  position: absolute;
  width: 0;
  border-left: 31px solid transparent;
  border-right: 31px solid transparent;
}

.c0:before {
  bottom: 100%;
  border-bottom: 17.89838337182448px solid #dedeeb;
}

.c0:after {
  top: 100%;
  width: 0;
  border-top: 17.89838337182448px solid #dedeeb;
}

.c0 .content {
  position: relative;
  top: -1px;
}

.c0 .content .icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .content .header {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c0 .content .react-flow__handle-bottom {
  top: 52.69515011547344px;
  z-index: 2;
}

.c1 {
  left: 1px;
  z-index: 1;
  position: relative;
  height: 35.64203233256351px;
  width: 60px;
  background-color: #f9fafb;
  margin: 17.821016166281755px 0;
}

.c1:before,
.c1:after {
  content: '';
  position: absolute;
  width: 0;
  border-left: 30px solid transparent;
  border-right: 30px solid transparent;
}

.c1:before {
  bottom: 100%;
  border-bottom: 16.821016166281755px solid #f9fafb;
}

.c1:after {
  top: 100%;
  width: 0;
  border-top: 16.821016166281755px solid #f9fafb;
}

.c1 .content {
  position: relative;
  top: -1px;
}

.c1 .content .icon {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .content .header {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c1 .content .react-flow__handle-bottom {
  top: 51.46304849884527px;
  z-index: 2;
}

<div
    className="c0 exagon"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    size="small"
  >
    <div
      className="c1 exagon inner"
      size="small"
    >
      <div
        className="content"
      >
        <div
          className="c2 header"
        >
          <div
            className="left as-column"
          >
            <div
              className="icon"
            >
              <div
                className="c3 svg-icon alert  "
              >
                <svg
                  height="20px"
                  width="20px"
                >
                  ico-alert.svg
                </svg>
              </div>
            </div>
            <div
              className="c4 header-text"
              size="small"
            >
              <div
                className="name"
              />
            </div>
          </div>
        </div>
        <div
          className=""
        />
      </div>
    </div>
  </div>,
  "",
]
`;
